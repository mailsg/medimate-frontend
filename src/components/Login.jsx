import React, { Component } from 'react';
import axios from 'axios';
import { NavLink } from 'react-router-dom';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      error: '',
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    const { username } = this.state; try {
      const response = await axios.post('http://localhost:3000/users/sign_in', {
        username,
      }); localStorage.setItem('token', response.data.token);
    } catch (error) {
      this.setState({ error: 'Invalid username' });
    }
  };

  render() {
    const { username, error } = this.state;
    return (
      <div className="login-container">
        <h2 className="heading">Login</h2>
        <form onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor="username">Username</label>
            <input
              type="text"
              id="username"
              name="username"
              value={username}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <button type="submit">Login</button>
          </div>
          <p>Dont have an account? Create one below</p>
          <div>
            <NavLink to="signup">
              <button type="submit">Signup</button>
            </NavLink>
          </div>
          {error && <p className="error-message">{error}</p>}
        </form>
      </div>
    );
  }
} export default Login;
