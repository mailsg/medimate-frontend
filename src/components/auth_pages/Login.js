import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import { useForm } from 'react-hook-form';
import styles from '../../css/reserve-form.module.css';

function LogIn() {
  const { reset } = useForm();
  const navigate = useNavigate();

  const [data, setData] = useState({
    email: '',
    password: '',
  });

  const [error, setError] = useState(null);

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!data.email || !data.password) {
      toast.warn('Please fill in all fields');
    }
    try {
      const response = await fetch('http://localhost:3000/users/sign_in', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          user: {
            email: data.email,
            password: data.password,
          },
        }),
      });

      if (response.ok) {
        const token = response.headers.get('Authorization');
        if (token) {
          toast.success(
            'You logged in successfully',
          );
          reset();
          localStorage.setItem('token', token);
          navigate('/');
        } else {
          setError('Authentication failed. Please try again.');
        }
      } else {
        const errorData = await response.json();
        setError(errorData.message);
      }
    } catch (error) {
      toast.error(
        'An error occured while creating the account, please try again',
      );
      reset();
    }
  };

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setData({
      ...data,
      [name]: value,
    });
  };

  return (
    <div className={styles['reservation-container']}>
      <h1 className={styles['reserve-form-header']}>Sign In with Email and Password</h1>
      <form onSubmit={handleSubmit} className={styles['reservation-form']}>
        <div className={styles['form-group']}>
          <input
            placeholder="Email"
            type="email"
            name="email"
            value={data.email}
            onChange={handleInputChange}
            // required
          />
        </div>
        <div className={styles['form-group']}>
          <input
            placeholder="Password"
            type="password"
            name="password"
            value={data.password}
            onChange={handleInputChange}
            // required
          />
        </div>
        {error && <p className="error">{error}</p>}
        {' '}
        <div className={styles['btn-container']}>
          <button type="submit" className={styles['submit-button']}>
            Log in
          </button>
        </div>
      </form>
      <div className={styles['float-btn-container']}>
        <p>Not registered?</p>
        <Link to="/sign_up">
          <button className={[styles['float-btn'], styles['submit-button']].join(' ')} type="submit">Sign up</button>
        </Link>
      </div>
    </div>
  );
}

export default LogIn;
