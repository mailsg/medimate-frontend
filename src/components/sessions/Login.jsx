import React, { Component } from 'react';
import axios from 'axios';
import { NavLink } from 'react-router-dom';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      error: '',
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    try {
      const response = await axios.post('http://localhost:3000/users/sign_in', {
        email,
        password,
      });

      const jtiToken = response.data.jti;

      localStorage.setItem('jtiToken', jtiToken);
    } catch (error) {
      this.setState({ error: 'Invalid email or password' });
    }
  };

  render() {
    const { email, password, error } = this.state;
    return (
      <div className="login-container">
        <h2 className="heading">Login</h2>
        <form onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor="email">Email</label>
            <input
              type="email"
              id="email"
              name="email"
              value={email}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <label htmlFor="password">Password</label>
            <input
              type="password"
              id="password"
              name="password"
              value={password}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <button type="submit">Login</button>
          </div>
          <p>
            Dont have an account?
            <NavLink to="signup">SignUp</NavLink>
          </p>
          {error && <p className="error-message">{error}</p>}
        </form>
      </div>
    );
  }
}

export default Login;
