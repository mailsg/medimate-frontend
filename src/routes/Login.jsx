// LoginPage.js
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import axios from 'axios';
import { loginSuccess } from '../redux/slice/authSlice';

const LoginPage = () => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = async () => {
    try {
      const response = await axios.post('http://localhost:3000/users/sign_in', {
        email,
        password,
      });
      // Assuming your backend returns user data upon successful login
      const userData = response.data.user;
      // Dispatch the action to store the user's information in Redux
      dispatch(loginSuccess(userData));

      // Redirect to the homepage or perform any other desired actions
    } catch (error) {
      // Handle login failure, display error messages, etc.
      console.error('Login failed', error);
    }
  };

  return (
    <div>
      <h1>Login Page</h1>
      <input
        type="text"
        placeholder="Email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <input
        type="password"
        placeholder="Password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <button type="button" onClick={handleLogin}>Login</button>
    </div>
  );
};

export default LoginPage;
