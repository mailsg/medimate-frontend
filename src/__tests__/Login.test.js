import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Login from '../components/auth_pages/Login';

describe('Login component', () => {
  it('Should render the component correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
