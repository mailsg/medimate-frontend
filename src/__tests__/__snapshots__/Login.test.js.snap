// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should render the component correctly 1`] = `
<div>
  <div
    class="reservation-container"
  >
    <h1
      class="reserve-form-header"
    >
      Sign In with Email and Password
    </h1>
    <form
      class="reservation-form"
    >
      <div
        class="form-group"
      >
        <input
          name="email"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
       
      <button
        class="submit-button"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;
